src/units/abilities.cpp

```diff
@@ -476,6 +476,54 @@ bool unit::has_ability_type(const std::string& ability) const
 	return !abilities_.child_range(ability).empty();
 }
 
+//this string abilitie is used for purpose of add a second halo if condition is matched.
+//It is put here header declaration commodity.
```

Comment:
I'm not really sure what this comment means. It looks like this function, which is called `create_vector`, doesn't actually create a vector - rather it adds a single element to an existing vector if it doesn't contain `tag_name` (which is actually an attribute)?

Reply:
function name changed in https://github.com/wesnoth/wesnoth/pull/7856/commits/ac14a495a281de959f9e2f4a6a931bf4a29acc1d, and tag_ame changed to 'image_type.